daemon/vpn/nordlynx/libtelio/libtelio.go

```diff
@@ -498,37 +455,43 @@ func (l *Libtelio) Refresh(c mesh.MachineMap) error {
 		return nil
 	}
 
-	meshnetMap := string(c.Raw)
+	config, err := teliogo.DeserializeMeshnetConfig(string(c.Raw))
+	if err != nil {
+		return fmt.Errorf("failed to deserialize meshnet config: %w", err)
+	}
 
-	result := teliogo.TELIORESOK
 	for i := 0; i < 10; i++ {
-		if result = l.lib.SetMeshnet(meshnetMap); result == teliogo.TELIORESOK {
+		if err = l.lib.SetMeshnet(config); err == nil {
 			break
 		}
 		time.Sleep(time.Millisecond * 100)
 	}
 
-	if result != teliogo.TELIORESOK {
-		return fmt.Errorf("failed to refresh meshnet: %d", result)
+	if err != nil {
+		return fmt.Errorf("failed to refresh meshnet %w", err)
 	}
 
-	l.meshnetMap = meshnetMap
+	l.meshnetConfig = config
 
 	return nil
 }
 
 type peer struct {
-	PublicKey string `json:"public_key"`
-	State     string `json:"state"`
+	PublicKey string
+	State     string
 }
 
 func (l *Libtelio) StatusMap() (map[string]string, error) {
 	l.mu.Lock()
 	defer l.mu.Unlock()
 
-	var peers []peer
-	if err := json.Unmarshal([]byte(l.lib.GetStatusMap()), &peers); err != nil {
-		return nil, fmt.Errorf("unmarshalling peer list: %w", err)
+	statusMap := l.lib.GetStatusMap()
+	peers := make([]peer, len(statusMap))
+	for i, node := range statusMap {
```

Comment:
This is kind of useless, it creates a vector that is used to create a map from it :smile: 
Maybe remove this and create directly the returned map?
And then the `type peer struct {` can be deleted?

Reply:
Haha, you're right, lol :smile: , fixing